cmd/acra-translator/acra-translator.go

```diff
@@ -162,6 +172,56 @@ func main() {
 		os.Exit(0)
 	})
 
+	go sigHandlerSIGHUP.RegisterWithContext(mainContext)
+	sigHandlerSIGHUP.AddCallback(func() {
+		log.Infof("Received incoming SIGHUP signal")
+		log.Debugf("Stop accepting new connections, waiting until current connections close")
+		readerServer.StopListeners()
+
+		// Set env flag for forked process
+		if err := os.Setenv(GracefulEnv, "true"); err != nil {
+			log.WithError(err).Errorln("Unexpected error on os.Setenv, graceful restart won't work. Please check env variables, especially gracefulEnv")
+		}
```

Comment:
Do I understand it correctly that if this fails then we still go on with restarting AcraTranslator? It will not reuse the listening sockets and will try to reopen them. It will probably not work since this process is still active, but I guess we give it a chance? Why not just fail immediately like we do with all errors that follow?

I guess we can't bail out at this point since `readerServer` has been already stopped, so our best bet is to die with an error and rely on service manager to restart AcraTranslator.

(Also, I believe the log message should use `GracefulEnv` constant value, not just `gracefulEnv` string.)

Reply:
Yes. It will use `Start` function without using descriptors. And actually I don't know. I ported it from our AcraServer (https://github.com/cossacklabs/acra/blob/master/cmd/acra-server/acra-server.go#L391). cc @Lagovas

Reply:
ok. Will change on `Fatalf`